docs/auth.md

```diff
@@ -0,0 +1,462 @@
+Host: cozy.example.org
+```
+
+**Note** we follow the TOFU principle (Trust On First Use). It means that if
+the user has already said yes for this authorization and scopes, she will be
+redirected to the app directly. As for `/auth/login`, the fragment is
+overriden in the redirection with `#_=_`.
+
+### POST /auth/authorize
+
+When the user accepts, her browser send a request to this endpoint:
+
+```http
+POST /auth/authorize
+Host: cozy.example.org
+Content-type: x-www-form-urlencoded
+
+approve=Approve&csrf-token=johw6Sho
+```
+
+**Note**: this endpoint is protected against CSRF attacks.
+
+The user is then redirected to the original client, with an access code in the
+URL:
+
+```http
+HTTP/1.1 302 Moved Temporarily
+Location: https://client.org/?state=Eh6ahshepei5Oojo&access_code=Aih7ohth
+```
+
+### POST /auth/access_token
+
+Now, the client can check that the state is correct, and if it is the case,
+ask for an `access_token`. It can use this route with the `code` (ie
+access_code) given below.
+
+This endpoint is also used to refresh the access token, by sending the
+`refresh_token` instead of the `access_code`.
+
+The parameters are:
+
+- `grant_type`, with `authorization_code` or `refresh_token` as value
+- `code` or `refresh_token`, depending on which grant type is used
+- `client_id`
+- `client_secret`
+
+Example:
+
+```http
+POST /auth/access_token
+Host: cozy.example.org
+Content-type: x-www-form-urlencoded
+Accept: application/json
+
+grant_type=authorization_code&code=Aih7ohth&client_id=oauth-client-1&client_secret=Oung7oi5
+```
+
+```http
+HTTP/1.1 200 OK
+Content-type: application/json
+
+{
+  "access_token": "ooch1Yei",
+  "token_type": "bearer",
+  "refresh_token": "ui0Ohch8",
+  "scope": "files/images:read data/io.cozy.contacts:read"
+}
+```
+
+### FAQ
+
+> What format is used for tokens?
+
+The tokens are formatted as [JSON Web Tokens (JWT)](https://jwt.io/).
+
+> What happens when the user has lost her password?
+
+She can reset it from the command-line, like this:
+
+```sh
+$ cozy-stack instances reset-password cozy.example.org
+ek0Jah1R
+```
+
+A new password is generated and print in the console.
+
+> Is two-factor authentication (2FA) possible?
+
+Yes, it's possible.
+
+**TODO:** explain how
+
+
+Client-side apps
+----------------
+
+**Important**: OAuth2 is not used here! The steps looks similar (like obtaining
+a token), but when going in the details, it doesn't match.
+
+### How to register the application?
+
+The application is registered at install. See [app management](apps.md) for
+details.
+
+### How to get a token?
+
+When a user access an application, she first loads the HTML page. Inside this
+page, the `<body>` tag has a `data-cozy-token` attribute with a token. This
+token is specific to a context, that can be either public or private.
```

Comment:
If we have a domain/app, why not cookies? There are easier to set from the server without parsing the HTML / requiring the app dev to put a placehoder and as easy to read from the javascript side. They also will apply to any subrequest without the need for the app developer to handle them manually.